### PR TITLE
Base36 byte-encoding specification

### PR DESCRIPTION
**Pull request by ribasushi**
As per our earlier decision we will start supporting base36 in all implementations, in order  to resolve https://github.com/ipfs/go-ipfs/issues/7318

This adds the encoding/specification to the table of codecs. Another PR fixing up the test vectors and adding an actual program to generate them will follow.

The encoding uses the alphabet 0-9a-z case insensitively. The prefix `K` is chosen
to limit future clashes with english words based on https://en.wikipedia.org/wiki/Letter_frequency#Relative_frequencies_of_the_first_letters_of_a_word_in_the_English_language ( also the prefix `K2` evokes lofty goals 😛 )

Typical encodings look like (generated by [this program](https://play.golang.org/p/sTBo5tkjLt_i)):
```
Nul Cid	lc: kdaznk
Nul Cid	UC: KDAZNK
Peer ID	lc: k51qzi5uqu5dj16qyiq0tajolkojyl9qdkr254920wxv7ghtuwcz593tp69z9m
Peer ID	UC: K51QZI5UQU5DJ16QYIQ0TAJOLKOJYL9QDKR254920WXV7GHTUWCZ593TP69Z9M
Raw UFS	lc: k2cwuec83ymcdya5erizp4urvb7ls2mytq6m2us1404b85f8qwk5tqtk
Raw UFS	UC: K2CWUEC83YMCDYA5ERIZP4URVB7LS2MYTQ6M2US1404B85F8QWK5TQTK
PB UFS	lc: k2jmtxs4gmv7e6433f6bula82cy8eg800nuu2yy7vrk4rd5u011bzxpf
PB UFS	UC: K2JMTXS4GMV7E6433F6BULA82CY8EG800NUU2YY7VRK4RD5U011BZXPF
```

/cc @aschmahmann who I can't add to reviewers

### CODE REVIEW

**Comment by Stebalien** [comment]
LGTM. Can we add a quick spec? We also need to specify that zero padding is done with zeros.

**Comment by ribasushi** [comment]
Nod, will write one once I finish go-multibase tests in a bit

**Comment by ribasushi** [comment]
@Stebalien https://github.com/multiformats/multibase/blob/07cd2adbc10a9f38b122120a51628d7ed3f46ffb/rfcs/Base36.md

File: multibase.csv [hunk 1]
```diff
@@ -5,17 +5,19 @@ base8,             7,    octal,
...
+base32hexpad,      t,    rfc4648 case-insensitive - with padding,                  candidate
+base32hexpadupper, T,    rfc4648 case-insensitive - with padding,                  candidate
+base32,            b,    rfc4648 case-insensitive - no padding,                    default
+base32upper,       B,    rfc4648 case-insensitive - no padding,                    default
+base32pad,         c,    rfc4648 case-insensitive - with padding,                  candidate
+base32padupper,    C,    rfc4648 case-insensitive - with padding,                  candidate
 base32z,           h,    z-base-32 (used by Tahoe-LAFS),                           draft
-base58flickr,      Z,    base58 flicker,                                           candidate
+base36,            k,    base36 [0-9a-z] case-insensitive - no padding,            default
+base36upper,       K,    base36 [0-9a-z] case-insensitive - no padding,            default
```
Comment (Stebalien) [inline multibase.csv:18]:
```suggestion
base36upper,       K,    base36 [0-9a-z] case-insensitive - no padding,            draft
```

(need wide implementation to be "default", even if we're planning on making it a default)

File: rfcs/Base36.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+The multibase base36 prefix is the character `k` or `K`. The digit-alphabet
+consists of 0..9 and then the case insensitive range a..z for the values 10..35
+
+## Encoding
+
+A byte array is encoded to base36 by:
+
+1. Counting the number of leading 0 bytes (Z).
+2. Interpreting the rest of the byte array as a big-endian unsigned integer (N).
+3. Concatenating a length Z string of '0' characters with the decimal
```
Comment (Stebalien) [inline rfcs/Base36.md:12]:
```suggestion
3. Concatenating a length Z string of '0' characters with the base36
```

File: multibase.csv (hunk 1 above)
Comment (ribasushi) [reply multibase.csv]: All fixed. Let's keep it open for a day until the rest of the stuff lands.

If anyone **REALLY** dislikes `K`: **SPEAK NOW!** 😜

**Comment by hugomrdias** [comment]
i will implement this in js https://github.com/multiformats/js-multibase/issues/53

**Pull request merged**